smart_open/s3.py

```diff
@@ -127,14 +134,23 @@ def open(
             resource_kwargs=resource_kwargs,
         )
     elif mode == WRITE_BINARY:
-        fileobj = BufferedOutputBase(
-            bucket_id,
-            key_id,
-            min_part_size=min_part_size,
-            session=session,
-            multipart_upload_kwargs=multipart_upload_kwargs,
-            resource_kwargs=resource_kwargs,
-        )
+        if multipart_upload:
+            fileobj = BufferedMultiPartOutputBase(
+                bucket_id,
+                key_id,
+                min_part_size=min_part_size,
+                session=session,
+                multipart_upload_kwargs=multipart_upload_kwargs,
+                resource_kwargs=resource_kwargs,
+            )
+        else:
+            fileobj = BufferedSinglePartOutputBase(
+                bucket_id,
+                key_id,
+                session=session,
+                multipart_upload_kwargs=multipart_upload_kwargs,
```

Comment:
Why are we passing multi-part arguments to the single-part writer?

Reply:
This should be `singlepart_upload_kwargs` to be passed onto the S3.Object.put method. Will change accordingly